File: solutions/fsharp/parallel-letter-frequency/1/ParallelLetterFrequency.fs
module ParallelLetterFrequency
open System

let freq (text : string) = 
    text.ToLower()
    |> Seq.filter Char.IsLetter
    |> Seq.countBy id

let frequency texts =
    [ for t in texts ->
        async {
            return freq t
        }
    ]
    |> Async.Parallel
    |> Async.RunSynchronously
    |> Seq.concat
    |> Seq.groupBy fst
    |> Seq.map (fun (char, count) -> (char, count |> Seq.sumBy snd))
    |> Map.ofSeq

File: solutions/csharp/parallel-letter-frequency/1/ParallelLetterFrequency.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ParallelLetterFrequency
{
    public static Dictionary<char, int> Calculate(IEnumerable<string> texts)
    {
        return texts
            .AsParallel()
            .SelectMany(text => text.ToLower().Select(c => c))
            .Where(c => char.IsLetter(c))
            .GroupBy(x => x)
            .ToDictionary(d => d.Key, d => d.Count());
    }
}